src/main/java/lumina/main/Lumina.java

```diff
@@ -0,0 +1,183 @@
+    private Storage storage;
+    private TaskList taskList;
+
+
+    // commands
+    private static final String ECHO_EXIT_STRING = "bye";
+    private static final String ECHO_LIST_STRING = "list";
+    private static final String ECHO_MARK_TASK_STRING = "mark";
+    private static final String ECHO_UNMARK_TASK_STRING = "unmark";
+    private static final String ECHO_TODO_TASK = "todo";
+    private static final String ECHO_DEADLINE_TASK = "deadline";
+    private static final String ECHO_EVENT_TASK = "event";
+    private static final String ECHO_DELETE_TASK = "delete";
+    private static final String ECHO_FIND_TASK = "find";
+
+
+    /**
+     * Constructor for the chatbot
+     *
+     * @throws IllegalStateException If already instantiated
+     */
+    public Lumina() {
+        if (luminaCount > 0) {
+            // since for now we only have one data file, we will prevent more instantiation
+            // of more than once Lumina instance, to prevent data from being corrupted
+            throw new IllegalStateException("Lumina has already been instantiated!");
+        }
+        luminaCount++;
+        ui = new Ui();
+        parser = new Parser();
+        storage = new Storage(parser);
+        taskList = new TaskList(ui, parser);
+
+        storage.loadData(taskList);
+    }
+
+    private void echo(Scanner sc) {
+        ui.greet();
+        String msg;
+        while(true) {
+            System.out.println();
+            msg = sc.nextLine();
+            
+            String command = msg.split(" ")[0].trim();
+
+            if (command.equals(Lumina.ECHO_EXIT_STRING)) {
+                storage.saveData(taskList);
+                ui.exit();
+                break;
+            }
+            if (command.equals(Lumina.ECHO_LIST_STRING)) {
+                taskList.listTasks();
+                continue;
+            }
+            // first check unmark since mark contains unmark
+            if (command.equals(Lumina.ECHO_UNMARK_TASK_STRING)) {
+                try {
+                    String[] msgSplit = msg.split(" ");
+                    if (msgSplit.length == 2) {
+                        int taskIndex = Integer.parseInt(msgSplit[1]) - 1; // 0 indexed
+                        taskList.markTaskNotDone(taskIndex);
+                    } else {
+                        throw new LuminaException("Oh no! Lumina detected "
+                                + "unexpected number of parameters in your command! "
+                                + "Please try again");
+                    }
+                } catch(LuminaException e) {
+                    ui.printMessage(e.getMessage());
+                }
+                continue;
+            }
+            if (command.equals(Lumina.ECHO_MARK_TASK_STRING)) {
+                try {
+                    String[] msgSplit = msg.split(" ");
+                    if (msgSplit.length == 2) {
+                        int taskIndex = Integer.parseInt(msgSplit[1]) - 1; // 0 indexed
```

Comment:
While msgSplit is understandable, perhaps renaming it to splitMessage or commandParts would make the code slightly more readable?